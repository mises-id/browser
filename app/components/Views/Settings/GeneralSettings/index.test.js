import React from 'react';
import {shallow} from 'enzyme';
import GeneralSettings from './';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';

const initialState = {
  privacy: {approvedHosts: [], privacyMode: true},
  browser: {history: []},
  settings: {
    lockTime: 1000,
    searchEngine: 'DuckDuckGo',
    useBlockieIcon: true,
  },
  engine: {
    backgroundState: {
      CurrencyRateController: {currentCurrency: 'USD'},
      NetworkController: {
        provider: {
          type: 'mainnet',
        },
      },
      PreferencesController: {selectedAddress: '0x0'},
    },
  },
};
const mockStore = configureMockStore();
const store = mockStore(initialState);

describe('GeneralSettings', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <GeneralSettings
          navigation={{
            state: {params: {}},
          }}
        />
      </Provider>,
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
