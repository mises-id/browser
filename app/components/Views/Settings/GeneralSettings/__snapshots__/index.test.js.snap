// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralSettings should render correctly 1`] = `
<Connect(Settings)
  navigation={
    Object {
      "state": Object {
        "params": Object {},
      },
    }
  }
/>
`;
